### PR TITLE
chore(web): move `common/web/input-processor/` → `web/src/engine/main/` 🏗️

### PR DESCRIPTION
**Pull request by ermshiperete**
Also:
- move `modelSpec.ts` and `PredictionContext.ts` → `web/src/engine/interfaces`
- move `correction.ts` and `correctionLayout.ts` to `web/src/engine/osk/`

Relates-to: #12035
Fixes: #12054

@keymanapp-test-bot skip

### CODE REVIEW

File: web/src/engine/osk/src/input/gestures/browser/subkeyPopup.ts [hunk 1]
```diff
@@ -6,7 +6,7 @@ import VisualKeyboard from '../../../visualKeyboard.js';
 import { DeviceSpec, ActiveSubKey, KeyDistribution, ActiveKeyBase } from '@keymanapp/keyboard-processor';
 import { ConfigChangeClosure, GestureRecognizerConfiguration, GestureSequence, PaddedZoneSource, RecognitionZoneSource } from '@keymanapp/gesture-recognizer';
 import { GestureHandler } from '../gestureHandler.js';
-import { CorrectionLayout, CorrectionLayoutEntry, distributionFromDistanceMaps, keyTouchDistances } from '@keymanapp/input-processor';
+import { CorrectionLayout, CorrectionLayoutEntry, distributionFromDistanceMaps, keyTouchDistances } from 'keyman/engine/osk';
```
Comment (jahorton) [inline web/src/engine/osk/src/input/gestures/browser/subkeyPopup.ts:9]: I feel it's a bit odd to have a class use an external-style import pattern for something within the same package, but maybe that's just me.  (This is hardly the only place with this sort of change.)

File: web/src/engine/main/src/text/prediction/languageProcessor.ts
Comment (jahorton) [inline web/src/engine/main/src/text/prediction/languageProcessor.ts:1]: If we're moving this, we may as well move it up to the `text` folder; no point in having it isolated within its own folder, personally.

File: web/src/engine/interfaces/src/prediction/languageProcessor.interface.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+ * Corresponds to the 'statechange' LanguageProcessor event.
+ */
+export type StateChangeHandler = (state: StateChangeEnum) => any;
+
+/**
+ * Covers 'tryaccept' events.
+ */
+export type TryUIHandler = (source: string, returnObj: { shouldSwallow: boolean }) => boolean;
+
+export interface LanguageProcessorEventMap {
```
Comment (jahorton) [inline web/src/engine/interfaces/src/prediction/languageProcessor.interface.ts:39]: Great way to move forward on this and preserve modularization for the different moving parts.  ❤️

File: web/src/engine/main/src/text/contextWindow.ts
Comment (jahorton) [inline web/src/engine/main/src/text/contextWindow.ts:1]: As these relocated pieces are headless-compatible, perhaps rename the folder `headless`?  Many parts of `engine/main` are not headless compatible, so it'd be good to clarify that distinction in some form.

**Review by jahorton — APPROVED** [review]
I appreciate you putting in the work on this one.  The `input-processor` package has long felt like it was kinda stuck in some sort of in-between limbo.  It was made in v12, long before the v17 modularization, and I do have to say, I like how the remapping of each piece looks to its new destination in the newer v17 modules.

File: web/src/engine/main/src/text/prediction/languageProcessor.ts
Comment (ermshiperete) [reply web/src/engine/main/src/text/prediction/languageProcessor.ts:1]: done.

**Review by mcdurdin — APPROVED** [review]

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 18.0.83-alpha](https://keyman.com/downloads/releases/alpha/18.0.83)

**Pull request merged**